src/libs/actions/OnyxUpdateManager.ts

```diff
@@ -27,6 +28,132 @@ Onyx.connect({
+            // so that we can continue searching for the next update after all gaps
+            gapExists = true;
+            firstUpdateAfterGaps = undefined;
+        }
+    }
+
+    let updatesAfterGaps: DeferredUpdatesDictionary = {};
+    if (gapExists && firstUpdateAfterGaps) {
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        updatesAfterGaps = Object.fromEntries(Object.entries(updates).filter(([lastUpdateID]) => Number(lastUpdateID) >= firstUpdateAfterGaps!));
+    }
+
+    let latestMissingUpdateID: number | undefined;
+    const updatesAfterGapValues = Object.values(updatesAfterGaps);
+    if (updatesAfterGapValues.length > 0 && updatesAfterGapValues[0]) {
+        latestMissingUpdateID = Number(updatesAfterGapValues[0].previousUpdateID) || 0;
+    }
+
+    return {applicableUpdates, updatesAfterGaps, latestMissingUpdateID};
+}
+
+// This function will check for gaps in the deferred updates and
+// apply the updates in order after the missing updates are fetched and applied
+function validateAndApplyDeferredUpdates(): Promise<void> {
+    // We only want to apply deferred updates that are newer than the last update that was applied to the client.
+    // At this point, the missing updates from "GetMissingOnyxUpdates" have been applied already, so we can safely filter out.
+    const pendingDeferredUpdates = Object.fromEntries(
+        Object.entries(deferredUpdates).filter(([lastUpdateID]) => {
+            // It should not be possible for lastUpdateIDAppliedToClient to be null,
+            // after the missing updates have been applied.
+            // If still so we want to keep the deferred update in the list.
+            if (!lastUpdateIDAppliedToClient) {
+                return true;
+            }
+            return (Number(lastUpdateID) ?? 0) > lastUpdateIDAppliedToClient;
+        }),
+    );
+    const pendingDeferredUpdateValues = Object.values(pendingDeferredUpdates);
+
+    // If there are no remaining deferred updates after filtering out outdated ones,
+    //  we can just unpause the queue and return
+    if (pendingDeferredUpdateValues.length === 0) {
+        return Promise.resolve();
+    }
+
+    const {applicableUpdates, updatesAfterGaps, latestMissingUpdateID} = detectGapsAndSplit(pendingDeferredUpdates);
+
+    // If we detected a gap in the deferred updates, only apply the deferred updates before the gap,
+    // re-fetch the missing updates and then apply the remaining deferred updates after the gap
+    if (latestMissingUpdateID) {
+        return new Promise((resolve, reject) => {
+            deferredUpdates = {};
+            applyUpdates(applicableUpdates).then(() => {
+                // Same as above, we can ignore this case because
+                // it should not be possible for lastUpdateIDAppliedToClient to be null.
+                if (!lastUpdateIDAppliedToClient) {
```

Comment:
also confirm if `latestMissingUpdateID > lastUpdateIDAppliedToClient`. Maybe parallel runs of other requests (like ReconnectApp) might have updated this variable